File: leetcode/questions/88/python/88.py
from typing import List

class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        nums2_limit = len(nums2)
        nums1_limit = len(nums1) - nums2_limit
        idx1 = nums1_limit - 1
        idx2 = nums2_limit - 1

        for idx_merged in reversed(range(len(nums1))):
            if idx2 >= 0 and (idx1 < 0 or nums1[idx1] <= nums2[idx2]):
                nums1[idx_merged] = nums2[idx2]
                idx2 -= 1
            elif idx2 < 0 or nums1[idx1] > nums2[idx2]:
                nums1[idx_merged] = nums1[idx1]
                idx1 -= 1

File: leetcode/questions/88/java/88.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int nums2Limit = nums2.length;
        int nums1Limit = nums1.length - nums2Limit;
        int idx1 = nums1Limit - 1;
        int idx2 = nums2Limit - 1; 

        for (int idxMerged = nums1.length - 1; idxMerged >= 0; --idxMerged) {
            if (idx2 >= 0 && (idx1 < 0 || nums1[idx1] <= nums2[idx2])) {
                nums1[idxMerged] = nums2[idx2];
                idx2 -= 1;
            }
            else if (idx2 < 0 || nums1[idx1] > nums2[idx2]) {
                nums1[idxMerged] = nums1[idx1];
                idx1 -= 1;
            }
        }
    }
}